### PR TITLE
Feature/update goss verifier

### PR DESCRIPTION
**Pull request by marcinpraczko**
Proposed solution for issue: #820

### CODE REVIEW

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 1]
```diff
@@ -5,33 +5,63 @@
...
+
+    - name: "Copy tests to remote"
+      copy:
+        src: "{{ playbook_dir }}/../files/goss/{{ item }}"
+        dest: "/tmp/{{ item }}"
       with_items:
         - hosts.yml
 
-- name: Execute goss tests
+- name: "Execute goss tests"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 2]
```diff
@@ -5,33 +5,63 @@
...
-      get_url: url=https://github.com/aelsabbahy/goss/releases/download/{{ cookiecutter.goss_version}}/goss-linux-amd64
-               dest=/usr/local/bin/goss
-               sha256sum={{ cookiecutter.goss_sha256sum }}
-               mode=0755
-{% raw %}
-    - name: Copy tests to remote
-      copy: src="{{ playbook_dir }}/../files/goss/{{ item }}"
-            dest="/tmp/{{ item }}"
+
+    - name: "Download and install goss"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 3]
```diff
@@ -5,33 +5,63 @@
...
+  vars:
+    goss_version: "{{ cookiecutter.goss_version }}"
+    goss_sha256sum: "{{ cookiecutter.goss_sha256sum }}"
+{%- raw %}
+    goss_url: "https://github.com/aelsabbahy/goss/releases/download/{{ goss_version }}/goss-linux-amd64"
+
   tasks:
-    - name: Install ca-certificates
-      apt: name=ca-certificates
+    - name: "Install ca-certificates"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 4]
```diff
@@ -5,33 +5,63 @@
 # The goss ansible module is installed with molecule.  The ANSIBLE_LIBRARY
 # path is updated appropriately on `molecule verify`.
 
-- name: Testing prerequisites
+# Details about ansible module:
+#  - https://github.com/indusbox/goss-ansible
+
+- name: "Testing prerequisites"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 5]
```diff
@@ -5,33 +5,63 @@
...
       with_items:
         - hosts.yml
 
-- name: Execute goss tests
+- name: "Execute goss tests"
   hosts: all
   gather_facts: no
+
+  vars:
+    goss_results: "goss_results"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 6]
```diff
@@ -5,33 +5,63 @@
...
-    - name: test goss files
-      goss: path="/tmp/{{ item }}"
-            format=documentation
+    - name: "Create foler for goss result files"
+      file:
+        path: "{{ goss_results }}"
+        state: directory
+        mode: 0700
+
+    - name: "Test goss files"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

File: molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml [hunk 7]
```diff
@@ -5,33 +5,63 @@
...
+      goss:
+        path: "/tmp/{{ item }}"
+        format: documentation
+        output_file: "{{ goss_results }}/{{ item }}.txt"
       with_items:
         - hosts.yml
 
+    - name: "Display details about goss results"
```
Comment (retr0h) [inline molecule/cookiecutter/verifier/goss/{{cookiecutter.repo_name}}/tests/test_default.yml]: Quotes are unnecessary here.

**Comment by retr0h** [comment]
Hi @marcinpraczko looks good with the exception of a few names having quotes around them.

**Comment by marcinpraczko** [comment]
Hi @retr0h, removed unnecessary quotes, will use gilt shortly for new version of goss library.

**Comment by marcinpraczko** [comment]
Hi @retr0h, code was re based and is ready to merge from my point of view.

**Comment by retr0h** [comment]
@marcinpraczko awesome you rock.  I will look at your changes shortly.

**Pull request merged**